docs/proposals/config_data.md

```diff
@@ -0,0 +1,327 @@
+# Generic Configuration Object
+
+## Abstract
+
+This proposal proposes a generic configuration object named ``ConfigData`` stored inside ``etcd`` that
+stores data used for the configuration of ``Kubernetes`` components.
```

Comment:
@bgrant0607 ack. We had a component focus back in march, I'll clean up those parts